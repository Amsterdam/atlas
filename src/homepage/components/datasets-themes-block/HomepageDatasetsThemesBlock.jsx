import React from 'react'
import classNames from 'classnames'
import Link from 'redux-first-router-link'
import {
  preserveQuery,
  toDatasets,
  toDatasetsWithFilter,
} from '../../../store/redux-first-router/actions'
import PARAMETERS from '../../../store/parameters'
import HomepageBlock from '../block/HomepageBlock'
import './HomepageDatasetsThemesBlock.scss'

const HomepageDatasetsThemesBlock = () => {
  const CATALOGUS_THEMES_CONFIG = [
    {
      name: 'Verkeer en infrastructuur',
      slug: 'verkeer-infrastructuur',
      icon: 'parkeren-beheer',
    },
    {
      name: 'Toerisme en cultuur',
      slug: 'toerisme-cultuur',
      icon: 'toerisme',
    },
    {
      name: 'Geografie',
      slug: 'geografie',
      icon: 'kaart',
    },
    {
      name: 'Bevolking',
      slug: 'bevolking',
      icon: 'bevolking',
    },
    {
      name: 'Openbare ruimte en groen',
      slug: 'openbare-ruimte-groen',
      icon: 'openbare-ruimte',
    },
    {
      name: 'Stedelijke ontwikkeling',
      slug: 'stedelijke-ontwikkeling',
      icon: 'stedelijke-ontwikkeling',
    },
    {
      name: 'Zorg en welzijn',
      slug: 'zorg-welzijn',
      icon: 'gezondheid-zorg-welzijn',
    },
    {
      name: 'Energie',
      slug: 'energie',
      icon: 'energie',
    },
  ]

  const columnTwo = [...CATALOGUS_THEMES_CONFIG]
  const columnOne = columnTwo.splice(0, Math.ceil(CATALOGUS_THEMES_CONFIG.length / 2))

  const columns = [{ id: 0, themes: columnOne }, { id: 1, themes: columnTwo }]

  return (
    <HomepageBlock
      classes="c-homepage__block--left c-homepage__block--datasets"
      linkAction={toDatasets()}
      title="Datasetcatalogus"
      description="Blader door datasets (verzamelingen gegevens)"
      blockIsLink={false}
    >
      <div className="c-catalogus-themes">
        <div className="u-grid">
          <div className="u-row">
            {columns.map(column => (
              <div key={column.id} className="u-col-sm--6">
                <div
                  className={`c-catalogus-themes__column ${classNames({
                    'c-catalogus-themes__column--left': column.id === 0,
                    'c-catalogus-themes__column--right': column.id === 1,
                  })}`}
                >
                  <ul>
                    {column.themes.map(theme => (
                      <li key={theme.slug}>
                        <Link
                          to={preserveQuery(toDatasetsWithFilter(), {
                            [PARAMETERS.FILTERS]: { groups: theme.slug },
                          })}
                          className={`qa-theme-link c-catalogus-themes__link c-catalogus-themes__icon c-catalogus-themes__icon--${
                            theme.icon
                          }`}
                        >
                          <div className="c-catalogus-themes__link__text">{theme.name}</div>
                        </Link>
                      </li>
                    ))}
                  </ul>
                </div>
              </div>
            ))}
          </div>
        </div>
      </div>
    </HomepageBlock>
  )
}

export default HomepageDatasetsThemesBlock
