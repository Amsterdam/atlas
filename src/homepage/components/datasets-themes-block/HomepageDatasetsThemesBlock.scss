@import "../../../shared/styles/config";

$base-padding: calc(#{$base-whitespace} * 2);

.c-catalogus-themes {
  padding: ($base-whitespace * .8) $base-padding $base-padding;

  &__column {
    font-size: 0;
    margin: 6px 0 17px; // to position the button links under the themes

    &--right {
      margin-left: 3 * $base-whitespace;
    }
  }

  &__icon {
    height: 40px;
    margin-bottom: ($base-whitespace / 2);
    width: 100%;

    &::before {
      display: block;
      float: left;
      margin-top: 7px;
    }

    a {
      color: $primary-dark;
      text-decoration: none;
    }

    &:hover,
    &:focus {
      a {
        color: $primary-contrast;
        text-decoration: underline;
      }
    }

    &--parkeren-beheer {
      @include icon-button-text-right("parkeren-beheer", "", "red");
    }

    &--toerisme {
      @include icon-button-text-right("toerisme", "", "red");
    }

    &--kaart {
      @include icon-button-text-right("kaart", "", "red");
    }

    &--bevolking {
      @include icon-button-text-right("bevolking", "", "red");
    }

    &--openbare-ruimte {
      @include icon-button-text-right("openbare-ruimte", "", "red");
    }

    &--stedelijke-ontwikkeling {
      @include icon-button-text-right("stedelijke-ontwikkeling", "", "red");
    }

    &--gezondheid-zorg-welzijn {
      @include icon-button-text-right("gezondheid-zorg-welzijn", "", "red");
    }

    &--energie {
      @include icon-button-text-right("energie", "", "red");
    }

    &--wonen-leefomgeving {
      @include icon-button-text-right("wonen-leefomgeving", "", "red");
    }

    &--werk-en-inkomen {
      @include icon-button-text-right("werk-en-inkomen", "", "red");
    }

    &--dienstverlening {
      @include icon-button-text-right("dienstverlening", "", "red");
    }

    &--bestuur-organisatie {
      @include icon-button-text-right("bestuur-organisatie", "", "red");
    }

    &--educatie-jeugd-diversiteit {
      @include icon-button-text-right("educatie-jeugd-diversiteit", "", "red");
    }
  }

  &__link {
    display: table;
    font-weight: $normal-weight;
    height: 40px;

    &:focus {
      color: $primary-contrast;
      text-decoration: underline;
    }

    &__text {
      display: table-cell;
      line-height: 20px;
      padding-right: $base-whitespace;
      vertical-align: middle;
      width: 100%;
    }
  }
}
