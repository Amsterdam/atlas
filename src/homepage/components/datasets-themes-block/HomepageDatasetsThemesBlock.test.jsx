import React from 'react'
import { shallow } from 'enzyme'

import HomepageDatasetsThemesBlock from './HomepageDatasetsThemesBlock'

describe('HomepageDatasetsThemesBlock.test', () => {
  it('should render', () => {
    const wrapper = shallow(<HomepageDatasetsThemesBlock />)

    expect(wrapper).toMatchSnapshot()
  })
})
