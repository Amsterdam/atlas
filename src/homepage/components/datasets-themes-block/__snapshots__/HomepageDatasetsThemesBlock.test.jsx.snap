// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomepageDatasetsThemesBlock.test should render 1`] = `
<HomepageBlock
  blockIsLink={false}
  classes="c-homepage__block--left c-homepage__block--datasets"
  description="Blader door datasets (verzamelingen gegevens)"
  hasTallDescription={false}
  linkAction={
    Object {
      "type": "atlasRouter/DATASETS",
    }
  }
  title="Datasetcatalogus"
>
  <div
    className="c-catalogus-themes"
  >
    <div
      className="u-grid"
    >
      <div
        className="u-row"
      >
        <div
          className="u-col-sm--6"
          key="0"
        >
          <div
            className="c-catalogus-themes__column c-catalogus-themes__column--left"
          >
            <ul>
              <li
                key="verkeer-infrastructuur"
              >
                <Connect(Link)
                  className="qa-theme-link c-catalogus-themes__link c-catalogus-themes__icon c-catalogus-themes__icon--parkeren-beheer"
                  data-test="themes-block-link"
                  to={
                    Object {
                      "meta": Object {
                        "additionalParams": Object {
                          "filters": Object {
                            "groups": "verkeer-infrastructuur",
                          },
                        },
                        "preserve": true,
                      },
                      "type": "atlasRouter/DATASETS",
                    }
                  }
                >
                  <div
                    className="c-catalogus-themes__link__text"
                  >
                    Verkeer en infrastructuur
                  </div>
                </Connect(Link)>
              </li>
              <li
                key="toerisme-cultuur"
              >
                <Connect(Link)
                  className="qa-theme-link c-catalogus-themes__link c-catalogus-themes__icon c-catalogus-themes__icon--toerisme"
                  data-test="themes-block-link"
                  to={
                    Object {
                      "meta": Object {
                        "additionalParams": Object {
                          "filters": Object {
                            "groups": "toerisme-cultuur",
                          },
                        },
                        "preserve": true,
                      },
                      "type": "atlasRouter/DATASETS",
                    }
                  }
                >
                  <div
                    className="c-catalogus-themes__link__text"
                  >
                    Toerisme en cultuur
                  </div>
                </Connect(Link)>
              </li>
              <li
                key="geografie"
              >
                <Connect(Link)
                  className="qa-theme-link c-catalogus-themes__link c-catalogus-themes__icon c-catalogus-themes__icon--kaart"
                  data-test="themes-block-link"
                  to={
                    Object {
                      "meta": Object {
                        "additionalParams": Object {
                          "filters": Object {
                            "groups": "geografie",
                          },
                        },
                        "preserve": true,
                      },
                      "type": "atlasRouter/DATASETS",
                    }
                  }
                >
                  <div
                    className="c-catalogus-themes__link__text"
                  >
                    Geografie
                  </div>
                </Connect(Link)>
              </li>
              <li
                key="bevolking"
              >
                <Connect(Link)
                  className="qa-theme-link c-catalogus-themes__link c-catalogus-themes__icon c-catalogus-themes__icon--bevolking"
                  data-test="themes-block-link"
                  to={
                    Object {
                      "meta": Object {
                        "additionalParams": Object {
                          "filters": Object {
                            "groups": "bevolking",
                          },
                        },
                        "preserve": true,
                      },
                      "type": "atlasRouter/DATASETS",
                    }
                  }
                >
                  <div
                    className="c-catalogus-themes__link__text"
                  >
                    Bevolking
                  </div>
                </Connect(Link)>
              </li>
            </ul>
          </div>
        </div>
        <div
          className="u-col-sm--6"
          key="1"
        >
          <div
            className="c-catalogus-themes__column c-catalogus-themes__column--right"
          >
            <ul>
              <li
                key="openbare-ruimte-groen"
              >
                <Connect(Link)
                  className="qa-theme-link c-catalogus-themes__link c-catalogus-themes__icon c-catalogus-themes__icon--openbare-ruimte"
                  data-test="themes-block-link"
                  to={
                    Object {
                      "meta": Object {
                        "additionalParams": Object {
                          "filters": Object {
                            "groups": "openbare-ruimte-groen",
                          },
                        },
                        "preserve": true,
                      },
                      "type": "atlasRouter/DATASETS",
                    }
                  }
                >
                  <div
                    className="c-catalogus-themes__link__text"
                  >
                    Openbare ruimte en groen
                  </div>
                </Connect(Link)>
              </li>
              <li
                key="stedelijke-ontwikkeling"
              >
                <Connect(Link)
                  className="qa-theme-link c-catalogus-themes__link c-catalogus-themes__icon c-catalogus-themes__icon--stedelijke-ontwikkeling"
                  data-test="themes-block-link"
                  to={
                    Object {
                      "meta": Object {
                        "additionalParams": Object {
                          "filters": Object {
                            "groups": "stedelijke-ontwikkeling",
                          },
                        },
                        "preserve": true,
                      },
                      "type": "atlasRouter/DATASETS",
                    }
                  }
                >
                  <div
                    className="c-catalogus-themes__link__text"
                  >
                    Stedelijke ontwikkeling
                  </div>
                </Connect(Link)>
              </li>
              <li
                key="zorg-welzijn"
              >
                <Connect(Link)
                  className="qa-theme-link c-catalogus-themes__link c-catalogus-themes__icon c-catalogus-themes__icon--gezondheid-zorg-welzijn"
                  data-test="themes-block-link"
                  to={
                    Object {
                      "meta": Object {
                        "additionalParams": Object {
                          "filters": Object {
                            "groups": "zorg-welzijn",
                          },
                        },
                        "preserve": true,
                      },
                      "type": "atlasRouter/DATASETS",
                    }
                  }
                >
                  <div
                    className="c-catalogus-themes__link__text"
                  >
                    Zorg en welzijn
                  </div>
                </Connect(Link)>
              </li>
              <li
                key="energie"
              >
                <Connect(Link)
                  className="qa-theme-link c-catalogus-themes__link c-catalogus-themes__icon c-catalogus-themes__icon--energie"
                  data-test="themes-block-link"
                  to={
                    Object {
                      "meta": Object {
                        "additionalParams": Object {
                          "filters": Object {
                            "groups": "energie",
                          },
                        },
                        "preserve": true,
                      },
                      "type": "atlasRouter/DATASETS",
                    }
                  }
                >
                  <div
                    className="c-catalogus-themes__link__text"
                  >
                    Energie
                  </div>
                </Connect(Link)>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</HomepageBlock>
`;
